### PR TITLE
Add UART and Protobuf Example

### PR DESCRIPTION
**Pull request by ecourtois**
This PR only adds an example that shows the configuration and usage of a second UART,
and uses Protobuf (nanopb package) messages to exchange data over this UART.
A client written in Python is also provided.

### CODE REVIEW

File: examples/uart-nanopb/main.cpp [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+#define COLOR_MACRO(name, value) #name,
+#include "colors.inc"
+#undef COLOR_MACRO
+};
+
+static void uart_rx_cb(void *arg, uint8_t data)
+{
+    (void)arg;
+    if (msg_length_bytes_received < msg_length_bytes_number) {
+        ((uint8_t *)&msg_length)[msg_length_bytes_received++] = data;
```
Comment (gdoffe) [inline examples/uart-nanopb/main.cpp:73]: Why is it called `msg_length` as it is a data buffer ??

File: examples/uart-nanopb/main.cpp [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+        ringbuffer_add_one(&(ctx.rx_buf), data);
+        msg_bytes_received++;
+    }
+
+    if (msg_bytes_received == msg_length) {
+        msg_t msg;
+        msg.content.value = msg_length;
+        msg_send(&msg, reader_pid);
+        msg_length_bytes_received = 0;
+    }
```
Comment (gdoffe) [inline examples/uart-nanopb/main.cpp:88]: I understand nothing to this callback, can you please add some comments ?

File: examples/uart-nanopb/main.cpp [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+    char rx_mem[PB_Response_size];
+    ringbuffer_t rx_buf;
+} uart_ctx_t;
+
+static uart_ctx_t ctx;
+
+static uint8_t msg_length_bytes_number = 4;
+static uint8_t msg_length_bytes_received = 0;
+static uint32_t msg_length = 0;
+static uint32_t msg_bytes_received = 0;
```
Comment (gdoffe) [inline examples/uart-nanopb/main.cpp:36]: Add some comments here please, to know the difference between all these variables.

**Review by gdoffe — CHANGES_REQUESTED** [review]

**Comment by ecourtois** [comment]
Close this PR as nanopb as finally been replaced by EmbeddedProto.
See #51

**Pull request closed without merging**